### DISCUSSION TITLE
Extra spaces in Authorization header

### DISCUSSION THREAD

**Discussion by PalAditya** [Kong/kong#9917] [Help] [answered]
We have seen a case where the Authorization header received by Kong has extra spaces - as in, the header is   
"**Authorization: Bearer&nbsp;&nbsp;abc**" and not "**Authorization: Bearer abc**"
The simplest fix would be to ensure there are no extra spaces, but in case that's not possible, does anyone have any idea how to handle this? openid-connect currently performs no such normalization.
One possibility I can think of is to use pre-function and normalize the header on our own.

**Comment by samugi** [comment, marked as answer]
I believe it would be best practice to deny access in case the header's value does not conform to the expected syntax. In case an alternative behaviour (like the trimming you describe) is required, then a pre-function should be an appropriate solution.

**Reply by PalAditya** [reply]
Thanks @samugi . We did end up using trimming (since the other input was not controllable) and it works fine for our use case. Leaving this here in case it helps others.
